src/collectors/loggly/loggly.py

```diff
@@ -0,0 +1,268 @@
+# coding=utf-8
+
+"""
+Collect numeric value of a field from loggly search results.
+
+This collector will retrieve the values of a configured field from loggly API
+and publish each one on the log's specified timestamp.
+
+The collector will limit the query to a timeframe equal to the collector's
+duration back from now.
+
+#### Dependencies
+
+    * requests
+    * futures
+
+Example config file LogglyCollector.conf
+
+```
+enabled = true
+interval = 60
+
+[ metric ]
+[[ test.metric ]]
+subdomain = example
+username = loggly_user
+password = loggly_pass
+field = some.field
+query = tag:some_tag AND some_text_field='VALUE'
+```
+
+Example config file LogglyCollector.conf using split configuration
+
+```
+enabled = true
+interval = 60
+
+[ metric ]
+path = "/etc/diamond/configs/loggly/"
+```
+
+Example metric file for split configuration
+
+```
+[ metric ]
+[[ test.metric ]]
+subdomain = example
+username = loggly_user
+password = loggly_pass
+field = some.field
+query = '*'
+```
+"""
+
+import diamond.collector as collector
+import concurrent.futures
+import configobj
+import datetime
+import requests
+import bunch
+import time
+import os
+
+from math import ceil, log10
+from diamond.metric import Metric
+from diamond.error import DiamondException
+
+
+def multi_getattr(obj, attr, **kw):
+    """Retrieve nested attributes."""
+    attributes = attr.split(".")
+    for i in attributes:
+        try:
+            obj = getattr(obj, i)
+            if callable(obj):
+                obj = obj()
+        except AttributeError:
+            if 'default' in kw:
+                return kw['default']
+            else:
+                raise
+    return obj
+
+
+class LogglyCollector(collector.Collector):
+
+    """Loggly Collector."""
+
+    def get_default_config_help(self):
+        """Update configuration help."""
+        config_help = super(
+            LogglyCollector, self).get_default_config_help()
+        config_help.update({
+            'id': 'Metric identifier',
+            'field': 'Name of the field to retrieve, required',
+            'count': 'Whether to retrieve the count of matching events',
+            'timestamp_field': 'Name of the timestamp field.'
+                               'Defaults to \'timestamp\'',
```

Comment:
Just for readability, does it make sense to double quote the string so you don't have to escape the single quote? If you choose to do so, please keep on the strings in the dictionary consistent, by making them use double quotes for the values (you don't have to change all the strings in the file).